typed-racket-lib/typed-racket/rep/type-rep.rkt

```diff
@@ -579,6 +579,17 @@
   [#:frees (λ (f) (make-invariant (f value)))]
   [#:key 'continuation-mark-key])
 
+;; Newtype
+;; comes from define-new-subtype
+;; nm: a symbol representing the name of the Newtype
+;; id: a symbol created with gensym
+;; ty: a type for the representation
+(def-type Newtype ([nm symbol?] [id symbol?] [ty Type/c])
```

Comment:
Only if `(string=? (new-cons a b) (new-cons a b))`, `(string=? (new-string a ...) (new-string a ...))`, and `(equal? (new-let ([x a]) b) (new-let ([x a]) b))` produce false. But if according to a `type=?` predicate  the `a` and `b` types are equal if `a` is a subtype of `b` and `b` is a subtype of `a` (see [this](https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/types/subtype.rkt#L697)), then since two types generated by different `define-new-subtype` declarations with the same base type, then those two types won't be the same type, and they won't even be related by sub-typing, so they are "new" (as in "fresh" or "distinct").